core/change.go

```diff
@@ -100,85 +101,132 @@ func (change Change) Write(writer io.Writer) error {
 	return err
 }
 
-func kindFromLabel(config Config, label string) *KindConfig {
-	for _, kindConfig := range config.Kinds {
-		if kindConfig.Label == label {
-			return &kindConfig
-		}
+// AskPrompts will ask the user prompts based on the configuration
+// updating the change as prompts are answered.
+func AskPrompts(change *Change, config Config, stdinReader io.ReadCloser) error {
+	p := Prompter{
+		change:      change,
+		config:      config,
+		stdinReader: stdinReader,
+		kind:        nil,
+		err:         nil,
 	}
 
-	panic("label not part of any kind")
+	return p.Prompt()
 }
 
-// AskPrompts will ask the user prompts based on the configuration
-// updating the change as prompts are answered.
-func AskPrompts(change *Change, config Config, stdinReader io.ReadCloser) error {
-	var (
-		err  error
-		kind *KindConfig
-	)
-
-	if len(config.Components) > 0 {
-		compPrompt := promptui.Select{
-			Label: "Component",
-			Items: config.Components,
-			Stdin: stdinReader,
-		}
+type Prompter struct {
+	change      *Change
+	config      Config
+	stdinReader io.ReadCloser
+	kind        *KindConfig
+	err         error
+}
 
-		_, change.Component, err = compPrompt.Run()
-		if err != nil {
-			return err
-		}
+func (p *Prompter) Prompt() error {
+	p.promptForComponent()
```

Comment:
These methods should return error and early return instead of checking if p.err is nil or not in each of them.

Reply:
done